Hush/Assets/Scripts/Keys/KeyCounterUI.cs

```diff
@@ -0,0 +1,18 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+[RequireComponent(typeof(Text))]
+public class KeyCounterUI : MonoBehaviour
+{
+    private Text keyCountText;
+
+    void Awake () {
+        keyCountText = GetComponent<Text>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        keyCountText.text = "Keys: " + KeyCollect.NumOfKeys.ToString();
```

Comment:
See #43 to see comments about the GameState object! Will affect you too and will make both scripts a little cleaner